### PR TITLE
3538: Add tests for API calls

### PR DESCRIPTION
**Pull request by kat1906**

### CODE REVIEW

File: app/pages/adduser/confirmation/controller.js [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import manifest from './manifest.json';
 import { addParamsToManifest } from '../../../common/helpers/addParamsToManifest';
 
-export const getAddUserConfirmationContext = () => ({
+export const getAddUserConfirmationContext = ({ params, organisationId }) => ({
```
Comment (alibombaye) [inline app/pages/adduser/confirmation/controller.js]: why the params? can't userAdded just be passed in instead

File: app/pages/adduser/confirmation/manifest.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
   "backLinkText": "Back to dashboard",
-  "title": "{{username}} account added",
+  "backLinkHref": "/organisations/{{organisationId}}",
```
Comment (alibombaye) [inline app/pages/adduser/confirmation/manifest.json]: We might want to be careful here with the relative URLS and it working in the cluster. Think we just want to go down a level

File: app/pages/adduser/confirmation/controller.js (hunk 1 above)
Comment (kat1906) [reply app/pages/adduser/confirmation/controller.js]: because it is passed to the `addParamsToManifest` so it made sense

Comment (kat1906) [reply app/pages/adduser/confirmation/controller.js]: incase we add more 🤷‍♀

File: app/pages/adduser/confirmation/manifest.json (hunk 2 above)
Comment (kat1906) [reply app/pages/adduser/confirmation/manifest.json]: it doesn't work consistently if you do that - I was having problems

File: app/pages/adduser/controller.js [hunk 3]
```diff
@@ -3,29 +3,32 @@ import { getContext } from './contextCreator';
...
+    logger.info(`User added: ${JSON.stringify(data)}`);
+    return { success: true, userAdded: `${data.firstName} ${data.lastName}` };
   } catch (err) {
-    if (err.response.status === 400) {
+    if (err.response && err.response.status === 400 && err.response.data) {
       return err.response.data;
     }
-    logger.error(err);
-    throw err;
+    logger.error(JSON.stringify(err));
```
Comment (alibombaye) [inline app/pages/adduser/controller.js]: How about we add this logger line to the error handler so we always log the error.

File: app/pages/adduser/ui.test.js [hunk 4]
```diff
@@ -161,9 +161,29 @@ test('should render add user button', async (t) => {
...
+  const phoneNumber = Selector('[data-test-id="question-phoneNumber"] input');
+  const emailAddress = Selector('[data-test-id="question-emailAddress"] input');
+  const addUserButton = Selector('[data-test-id="add-user-button"] button');
+
+  await t
+    .expect(addUserButton.exists).ok()
+    .typeText(firstName, 'Peter')
+    .typeText(lastName, 'Parker')
+    .typeText(phoneNumber, '07777777777')
+    .typeText(emailAddress, 'Peter@Parker.com')
```
Comment (alibombaye) [inline app/pages/adduser/ui.test.js]: Upto you but we don't really need to type the text into each field. We don't do any validation on the node side. So it being empty is also perfectly fine.

Comment (kat1906) [reply app/pages/adduser/ui.test.js]: it looks nice when it types it in 😄

Comment (kat1906) [reply app/pages/adduser/ui.test.js]: I remember now - I wondered why I couldn't find an example haha

File: app/pages/adduser/controller.js (hunk 3 above)
Comment (kat1906) [reply app/pages/adduser/controller.js]: we need to have a whole discussion around error handling as it is inconsistent and even missing in places but I have removed this for now

File: app/pages/adduser/confirmation/manifest.json (hunk 2 above)
Comment (kat1906) [reply app/pages/adduser/confirmation/manifest.json]: might need a wider discussion

File: app/pages/organisation/controller.test.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+      firstName: 'Some',
+      lastName: 'Body',
+      phoneNumber: '07777777777',
+      emailAddress: 'some@body.com',
+      isDisabled: false,
+    },
+  ],
+};
+
+describe('dashboard controller', () => {
```
Comment (alibombaye) [inline app/pages/organisation/controller.test.js]: should this be Organisation controller?

File: app/pages/adduser/confirmation/controller.js (hunk 1 above)
Comment (alibombaye) [reply app/pages/adduser/confirmation/controller.js]: I get it now. I still think this should be passed though as userAdded here and we don't need organisationId.

File: app/pages/adduser/confirmation/manifest.json (hunk 2 above)
Comment (alibombaye) [reply app/pages/adduser/confirmation/manifest.json]: Ok let's discuss tomorrow.

File: app/pages/dashboard/contextCreator.js [hunk 6]
```diff
@@ -9,4 +10,5 @@ export const getContext = ({ organisations }) => ({
       { data: row.odsCode || '' },
     ])
     : [],
+  backLinkHref: `${baseUrl}/`,
```
Comment (alibombaye) [inline app/pages/dashboard/contextCreator.js]: 😢 I don't think this is right here. From the dashboard we click back we want to go back to public-browse.

Comment (kat1906) [reply app/pages/dashboard/contextCreator.js]: OH YEH GOOD SPOT 👀

Comment (alibombaye) [reply app/pages/dashboard/contextCreator.js]: Leave this as is. We'll come back to this in a separate ticket.

Comment (kat1906) [reply app/pages/dashboard/contextCreator.js]: I have changed it to '/' as `/admin/` doesn't go anywhere

Comment (kat1906) [reply app/pages/dashboard/contextCreator.js]: I have changed it to '/' as '/admin/' doesn't go anywhere

**Pull request merged**